### PR TITLE
feat(watch): press `r` should rerun current pattern tests

### PR DESCRIPTION
**Pull request by Dunqing**
### Why?

In debugging, we may need to change the `node_modules` files, but it does not automatically trigger a re-run of the test.

>  Besides, I think press `enter` is a better way to execute rerun. but the `enter` has been used in rerun all tests.

### CODE REVIEW

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Do you think this shortcut should be mentioned anywhere?

**Comment by Dunqing** [comment]
> Do you think this shortcut should be mentioned anywhere?

I searched for a long time without finding any documentation about these shortcuts, until I saw in the source code 😩

It looks like we need to add a `Watch Mode` page to the documentation and add the documentation related to shortcuts here.

**Review by AriPerkkio — COMMENTED** [review]
> Do you think this shortcut should be mentioned anywhere?

This should be mentioned here:

https://github.com/vitest-dev/vitest/blob/77d071a91f81b6faef28cb30c6c5b617a661459d/packages/vitest/src/node/stdin.ts#L7-L14

Also please add test case here if possible https://github.com/vitest-dev/vitest/blob/main/test/watch/test/stdin.test.ts

File: test/watch/test/stdin.test.ts [hunk 1]
```diff
@@ -17,6 +17,15 @@ test('quit watch mode', async () => {
   await vitest.isDone
 })
 
+test('rerun current pattern tests', async () => {
+  const pattern = 'TEST_NAME_PATTERN'
+  const vitest = await startWatchMode(`-t ${pattern}`)
```
Comment (AriPerkkio) [inline test/watch/test/stdin.test.ts:22]: `startWatchMode` accepts an array of strings. A single argument separated with spaces doesn't seem to work through `execa`. Let's change this to array instead.

In the `test/watch/fixtures` there is no test case that would match `TEST_NAME_PATTERN`. Let's change this to `sum` so it matches test case in `fixtures/math.test.ts`.

```suggestion
  const vitest = await startWatchMode('-t', 'sum')
```

File: test/watch/test/stdin.test.ts [hunk 2]
```diff
@@ -17,6 +17,15 @@ test('quit watch mode', async () => {
...
+
+  vitest.write('r')
+
+  await vitest.waitForOutput(`Test name pattern: ${pattern}`)
```
Comment (AriPerkkio) [inline test/watch/test/stdin.test.ts:26]: The output is formatted as Regexp:

```suggestion
  await vitest.waitForOutput('Test name pattern: /sum/')
```

File: test/watch/test/stdin.test.ts [hunk 3]
```diff
@@ -17,6 +17,15 @@ test('quit watch mode', async () => {
...
+})
```
Comment (AriPerkkio) [inline test/watch/test/stdin.test.ts:27]: Add one additional check to make sure `r` preserved the pattern properly. 

```suggestion
  await vitest.waitForOutput('1 passed')
})
```

File: packages/vitest/src/node/stdin.ts [hunk 4]
```diff
@@ -5,7 +5,8 @@ import { isWindows, stdout } from '../utils'
 import type { Vitest } from './core'
 
 const keys = [
-  ['a', 'rerun all tests'],
+  [['a, return'], 'rerun all tests'],
```
Comment (AriPerkkio) [inline packages/vitest/src/node/stdin.ts:8]: This breaks test run cancelling when pressing `'a'` as `cancelKeys` no longer contains `'a'`:

https://github.com/vitest-dev/vitest/blob/77d071a91f81b6faef28cb30c6c5b617a661459d/packages/vitest/src/node/stdin.ts#L15

**Review by AriPerkkio — CHANGES_REQUESTED** [review]

File: packages/vitest/src/node/stdin.ts (hunk 4 above)
Comment (Dunqing) [reply packages/vitest/src/node/stdin.ts:8]: Thanks!!

**Review by sheremet-va — APPROVED** [review]

**Review by AriPerkkio — APPROVED** [review]

**Pull request merged**